assets/src/edit-story/components/panels/stylePreset/utils.js

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import deepEqual from 'deep-equal';
+
+export function findMatchingColor(color, stylePresets, isText) {
+  // Ensure that the color doesn't already exist.
+  const colorsToMatch = isText ? stylePresets.textColors : stylePresets.colors;
+  return colorsToMatch.find((value) => deepEqual(color, value, true));
```

Comment:
Regarding concern about `deep-equal`: We can just use `JSON.stringify`? For small objects, it's rather quick. Alternatively, `fast-deep-equal` should come in at rather few KB, but still more of course.

Reply:
I was thinking of JSON.stringify originally, however, wasn't sure if it's guaranteed that the object values will always be in the same order from the color picker.

Also, could the logic change in the future? I suppose we could change it back to deep equality in the future if need be. 🤷‍♀ 

I have a preference towards not using stringify, however, no super strong opinion here.

WDYT, @swissspidy?

Reply:
We also have a simpler set of `generatePatternStyles` and CSS version of it. They are a bit better than `JSON.stringify` b/c we never need to worry about field ordering.

Reply:
Also see `useCommonColorValue`.

Reply:
Uh, that's a great suggestion. Yes, compare the generated CSS attribute and value. That'll also take care of slight differences in color object that will render the same. E.g. these will have similar outputs, yet different internals:

```js
const g1 = {
  type: 'linear',
  alpha: 1,
  stops: [
    { color: { r: 0, g: 0, b: 0, a: 0 }, position: 0 },
    { color: { r: 0, g: 0, b: 0, a: 0.5 }, position: 1 },
  ],
};
const g2 = {
  type: 'linear',
  alpha: 0.5,
  stops: [
    { color: { r: 0, g: 0, b: 0, a: 0 }, position: 0 },
    { color: { r: 0, g: 0, b: 0, a: 1 }, position: 1 },
  ],
};

Reply:
Great suggestion indeed, didn't think about it at all! Implemented now.